### ISSUE TITLE
Pagination pages 2...> not accessible using `tab` key tabindex #a11y

### ISSUE THREAD

**Issue by scottsasaki** [bootstrap-vue/bootstrap-vue#3496]
### Pagination is not accessible using `tab` key #a11y

Official bootstrap allows users to use the tab key on the keyboard to navigate to available pages. This differs from bootstrap-vue where only the current page is tabbable, then the user must change to using left/right keys on the keyboard. 

This could be difficult for someone with low mobility that has been using the tab key to navigate and then needs to transition to the arrow keys.

### Steps to reproduce the bug

1. Go to https://bootstrap-vue.js.org/docs/components/pagination/#pagination
2. Using the keyboard hit the tab key until your :focus lands on the example 1st page.
3. Hit tab again and you will be moved past the pagination box.

### Expected behavior

`tabindex should be set to 0`. The user should be able to tab to the next enabled page, as well as enabled left/right arrows.

### Versions 

v2.0.0-rc.22

**Libraries:**

- BootstrapVue: 2.0.0
- Bootstrap: 4.#.#

**Environment:**

- Device: Macbook
- OS: Mojave
- Browser: Chrome
- Version: 74

### Demo link

https://bootstrap-vue.js.org/docs/components/pagination/#pagination

### Additional context

I couldn't find any documentation around why the tabindex was set to `-1` when the page was enabled

**Comment by peoray** [comment]
Hello, I'd like to work on this :slightly_smiling_face:

**Comment by tmorehouse** [comment]
Pagination uses built in key board navigation.  the pagination component is a widget that has the current page item with `tabindex=0`, the other pages are reachable via left/right/home/end keys when using the keyboard.

https://bootstrap-vue.js.org/docs/components/pagination#keyboard-navigation-support

b-pagination follows the WAI-ARIA roving tab index pattern: https://www.w3.org/TR/wai-aria-practices-1.2/#kbd_roving_tabindex

**Comment by tmorehouse** [comment]
We would maybe add in a flag prop for disabling the WAI ARIA best practices keyboard navigation, which you would need to opt-in by setting the prop to true.
